pennylane/templates/subroutines/qpe.py

```diff
@@ -232,17 +235,16 @@ def error(self):
 
     # pylint: disable=protected-access
     def map_wires(self, wire_map: dict):
-        new_op = super().map_wires(wire_map)
```

Comment:
`Operation.map_wires` uses `copy.copy`, which is insufficient for `QuanutmPhaseEstimation`, because this leaves references to the original `unitary`, `target_wires` and `estimation_wires` hyperparameters. As a consequence, `map_wires` used to modify the original op, which is a bug.

We fox this by copying the relevant two lines from `Operation.map_wires` and replacing `copy.copy` by `copy.deepcopy`.

Reply:
😬

Reply:
We could also fix this in `Operation.map_wires`, of course...